### ISSUE TITLE
collapsible_if shouldn't apply if there is a comment before

### ISSUE THREAD

**Issue by Zoxc** [rust-lang-nursery/rust-clippy#798]
In this example there is comment which applies to both branches in the `else` block.

```
        if entry.end > entry.base {
            prev = entry_; // Go to the next entry
        } else {
            // No usable memory in this entry. Remove it from the list.

            if prev == st.list {
                st.list = entry.next;
                prev = st.list;
            } else {
                (*prev).next = entry.next;
            }
        }
```

**Comment by joshtriplett** [comment]
Similar example from rustc's tidy:

```rust
    // If this path is in-tree, we don't require it to be on the whitelist
    if must_be_on_whitelist {
        // If this dependency is not on the WHITELIST, add to bad set
        if !whitelist.contains(&krate.into()) {
            unapproved.insert(krate.into());
        }
    }
```

Both of the `if` statements have a comment, and collapsing them together would make this less clear.

**Comment by lukasstevens** [comment]
I'd like to work on this if there is still interest.

**Comment by flip1995** [comment]
Great, it's all yours. Also have a look at #3228, for some more discussion about this lint.
